common/djangoapps/student/views.py

```diff
@@ -2110,6 +2116,27 @@ def create_account_with_params(request, params):
     return new_user
 
 
+def _apply_third_party_auth_overrides(request, extra_fields):
+    """
+    Modify the required attribute of the provided extra_fields according to the learners third party auth provider.
+    """
+    updated_fields = deepcopy(extra_fields)
+
+    if third_party_auth.is_enabled():
+        running_pipeline = third_party_auth.pipeline.get(request)
+        if running_pipeline:
+            current_provider = third_party_auth.provider.Registry.get_from_pipeline(running_pipeline)
+            # Make input fields read-only for fields passed in via SSO callback data.
+            if current_provider and current_provider.sync_learner_profile_data:
+                for field_name, field_value in extra_fields.items():
+                    if field_name not in REGISTRATION_DISPLAY_FIELD_NAMES_FOR_THIRD_PARTY_AUTH and \
+                            field_value == 'required':
+                        # Make hidden fields optional
+                        updated_fields[field_name] = 'optional'
+
+    return updated_fields
```

Comment:
What is this code accomplishing? Why do we need it?

Reply:
This makes hidden fields optional so that learner can create an account, without this we get `"Field is required"` error. 
Registration page is displayed by a view in `student_account` app where we hide fields when `sync_learner_profile_data` is enabled and account creation handled by this API must conform with the field overrides there.

Reply:
I don't think we should be hiding fields that are required by the site and for which we don't have a value provided by the IdP.

Said another way, if the site requires certain user profile fields and the IdP does not provide values for these fields, we should be asking the user to enter values for these fields on the registration form.

Reply:
If we do that then fields like `country`, `education` may appear for most of IdPs and my understanding was that we wanted to show minimal set of fields to the learner at the registration page and create learner's account and then learner can add more info in the account settings page ?

Reply:
I think we still need to display any fields that are required by the site for which we have not received any values from the IdP. Maybe a good think to clarify at stand up today.